Crash report on app start

Build version: 1.7.1 
Current date: 2018-03-08 12:07:33 
Device: BullittGroupLimited S40 
 
Stack trace:  
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.health.openscale/com.health.openscale.gui.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.health.openscale.core.datatypes.ScaleUser.getId()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2325)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
	at android.app.ActivityThread.access$800(ActivityThread.java:151)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5255)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.health.openscale.core.datatypes.ScaleUser.getId()' on a null object reference
	at com.health.openscale.core.OpenScale.getTupleScaleData(OpenScale.java:226)
	at com.health.openscale.gui.fragments.OverviewFragment.updateOnView(OverviewFragment.java:167)
	at com.health.openscale.core.OpenScale.registerFragment(OpenScale.java:475)
	at com.health.openscale.gui.fragments.OverviewFragment.onCreateView(OverviewFragment.java:146)
	at android.support.v4.app.Fragment.performCreateView(Fragment.java:2261)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1419)
	at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1750)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1819)
	at android.support.v4.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3227)
	at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:3181)
	at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:192)
	at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:572)
	at android.support.v7.app.AppCompatActivity.onStart(AppCompatActivity.java:177)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1236)
	at android.app.Activity.performStart(Activity.java:6052)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2288)
	... 10 more

User actions:

--- a/android_app/app/src/main/java/com/health/openscale/core/OpenScale.java
+++ b/android_app/app/src/main/java/com/health/openscale/core/OpenScale.java
@@ -183,6 +183,10 @@ public class OpenScale {
             final int selectedUserId = getSelectedScaleUserId();
             if (selectedUserId != -1) {
                 selectedScaleUser = userDAO.get(selectedUserId);
+                if (selectedScaleUser == null) {
+                    selectScaleUser(-1);
+                    throw new Exception("could not find the selected user");
+                }
                 return selectedScaleUser;
             }
         } catch (Exception e) {
@@ -221,7 +225,7 @@ public class OpenScale {
 
     public ScaleMeasurement[] getTupleScaleData(int id)
     {
-        ScaleMeasurement[]  tupleScaleData = new ScaleMeasurement[3];
+        ScaleMeasurement[] tupleScaleData = new ScaleMeasurement[3];
 
         tupleScaleData[0] = measurementDAO.getPrevious(id, getSelectedScaleUser().getId());
         tupleScaleData[1] = measurementDAO.get(id);

